Dockerfile

```diff
@@ -14,57 +14,26 @@
 # Note that we're explicitly using the Debian bullseye image to make sure we're
 # compatible with the Python container we'll be copying the pathfinder
 # executable to.
-FROM rust:1.65-bullseye AS rust-builder
-
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y libssl-dev && rm -rf /var/lib/apt/lists/*
-
+FROM lukemathwalker/cargo-chef:0.1.48-rust-1.65.0-bullseye AS cargo-chef
 WORKDIR /usr/src/pathfinder
 
-# Build only the dependencies first. This utilizes
-# container layer caching for Rust builds
-RUN mkdir crates \
-    && cargo new --lib --vcs none crates/stark_curve \
-    && cargo new --lib --vcs none crates/stark_hash \
-    && cargo new --lib --vcs none crates/common \
-    && cargo new --lib --vcs none crates/ethereum \
-    && cargo new --lib --vcs none crates/gateway-client \
-    && cargo new --lib --vcs none crates/gateway-test-fixtures \
-    && cargo new --lib --vcs none crates/gateway-types \
-    && cargo new --lib --vcs none crates/retry \
-    && cargo new --lib --vcs none crates/serde \
-    && cargo new --lib --vcs none crates/storage \
-    && cargo new --lib --vcs none crates/pathfinder
-
-COPY Cargo.toml Cargo.lock ./
-
-COPY crates/stark_curve/Cargo.toml crates/stark_curve/Cargo.toml
-COPY crates/stark_hash/Cargo.toml crates/stark_hash/
-COPY crates/stark_hash/benches crates/stark_hash/benches
-
-COPY crates/common/Cargo.toml crates/common/build.rs crates/common/
-COPY crates/ethereum/Cargo.toml crates/ethereum/Cargo.toml
-COPY crates/gateway-client/Cargo.toml crates/gateway-client/Cargo.toml
-COPY crates/gateway-test-fixtures/Cargo.toml crates/gateway-test-fixtures/Cargo.toml
-COPY crates/gateway-types/Cargo.toml crates/gateway-types/Cargo.toml
-COPY crates/retry/Cargo.toml crates/retry/Cargo.toml
-COPY crates/serde/Cargo.toml crates/serde/Cargo.toml
-COPY crates/storage/Cargo.toml crates/storage/Cargo.toml
-
-COPY crates/pathfinder/benches crates/pathfinder/benches
-
 # refresh indices, do it with cli git for much better ram usage
 RUN CARGO_NET_GIT_FETCH_WITH_CLI=true cargo search --limit 0
 
-# DEPENDENCY_LAYER=1 should disable any vergen interaction, because the .git directory is not yet available
-RUN CARGO_INCREMENTAL=0 DEPENDENCY_LAYER=1 cargo build --release -p pathfinder
+FROM cargo-chef AS rust-planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
```

Comment:
I can't find the recipe file :eyes:

Reply:
Still a draft PR so I expect it will come in time

Reply:
The recipe file is generated here, and then copied to the actual build container a few lines below.